op-batcher/batcher/driver.go

```diff
@@ -699,7 +709,9 @@ func (l *BatchSubmitter) publishTxToL1(ctx context.Context, queue *txmgr.Queue[t
 
 	// Collect next transaction data. This pulls data out of the channel, so we need to make sure
 	// to put it back if ever da or txmgr requests fail, by calling l.recordFailedDARequest/recordFailedTx.
+	l.channelMgrMutex.Lock()
 	txdata, err := l.channelMgr.TxData(l1tip.ID())
+	l.channelMgrMutex.Unlock()
```

Comment:
I don't think that is worth it, here: we would need to declare txdata outside the anonymous fn and generally it's more boilerplate for no change in functionality.

Reply:
It would unlock the mutex in case of a panic, which is slightly better. But then, if we have a panic inside `TxData`, the batcher is broken. So it's fine for now. Just generally feeling better about always unlocking in a `defer`.